blocks/v2-resources-gallery/v2-resources-gallery.js

```diff
@@ -0,0 +1,113 @@
+import { wrapImageWithVideoLink, selectVideoLink, isVideoLink } from '../../scripts/video-helper.js';
+import { createElement } from '../../scripts/common.js';
+
+const blockName = 'v2-resources-gallery';
+
+export default function decorate(block) {
+  const blockHeading = block.querySelector('div:first-child');
+  blockHeading.classList.add(`${blockName}__heading`);
+
+  const viewAllButton = createElement('button', {
+    classes: ['v2-resources-gallery__toggle-resources', 'tertiary'],
+    props: {
+      type: 'button', title: 'Toggle resources list',
+    },
+  });
+
+  const documentRowsWrapper = createElement('div', {
+    classes: 'v2-resources-gallery__rows-wrapper',
+  });
+
+  const viewAllIcon = createElement('i', {
+    classes: 'v2-resources-gallery__toggle-resources-icon',
+  });
+
+  viewAllButton.innerText = 'view all';
+  viewAllButton.setAttribute('aria-expanded', 'false');
+
+  // Go through all rows
+  [...block.children].forEach((row) => [...row.children].forEach((elem) => {
+    // Add row and item classes accordingly
+    row.classList.add(`${blockName}__row`);
+    elem.classList.add(`${blockName}__item`);
+
+    // give p containing the image a specific class
+    const picture = elem.querySelector('picture');
+    if (picture && picture.closest('p')) picture.closest('p').classList.add('image');
+
+    const links = elem.querySelectorAll('a');
+    const videos = [...links].filter((link) => isVideoLink(link));
+
+    if (videos.length) {
+      // display image as link with play icon
+      const selectedLink = selectVideoLink(videos);
+      row.classList.add(`${blockName}__row--has-video`);
+      elem.classList.add(`${blockName}__item--has-video`);
+
+      if (selectedLink) {
+        picture.after(selectedLink);
+        wrapImageWithVideoLink(selectedLink, picture);
+      }
+
+      // remove all the videos links and exclude the selected one
+      videos.forEach((link) => link !== selectedLink && link.parentElement.remove());
+    } else {
+      row.classList.add(`${blockName}__row--has-documents`);
```

Comment:
While I see your point regarding semantics in this case I would like to keep it as it is for the following reasons:

1. This is not a perfect case for a list since we have more than one item in a list and helix won't let me group them all in one LI but it is wrapping each of them in one Li. 
It even renders a empty LI for a reason I cannot discover.
2. Having these items as list would be error prone for content authors and consequentially for us because if they improperly change a single line our js logic would be broken.
3. Of course we could bypass these things but I think we do not have time for these kind of semantic exercise, in other words, if we want to use LIs here: We gain nothing and we loose time and have error prone code.